src/main/kotlin/com/manimdsl/runtime/ASTExecutor.kt

```diff
@@ -210,9 +210,16 @@ class VirtualMachine(
             is LoopStatementNode -> executeLoopStatement(statement)
             is InternalArrayMethodCallNode -> executeInternalArrayMethodCall(statement)
             is StartCodeTrackingNode -> {
-                previousStepIntoState = stepInto
-                stepInto = statement.isStepInto
-                EmptyValue
+                val condition = executeExpression(statement.condition)
+                if (condition is BoolValue) {
+                    previousStepIntoState = stepInto
+                    if (condition.value) {
+                        stepInto = statement.isStepInto
+                    }
+                    EmptyValue
+                } else {
+                    condition
```

Comment:
Why do we return condition if it's not a bool value? Is it because of RuntimeErrors?

Reply:
yeah it wasn't implemented before